Source/VisualStudioTools/Private/BlueprintReferencesCommandlet.cpp

```diff
@@ -0,0 +1,220 @@
+#include "Policies/CondensedJsonPrintPolicy.h"
+#include "VisualStudioTools.h"
+
+namespace VisualStudioTools
+{
+static FString StripClassPrefix(const FString&& InClassName)
+{
+	if (InClassName.IsEmpty())
+	{
+		return InClassName;
+	}
+
+	size_t PrefixSize = 0;
+
+	const TCHAR ClassPrefixChar = InClassName[0];
+	switch (ClassPrefixChar)
+	{
+	case TEXT('I'):
+	case TEXT('A'):
+	case TEXT('U'):
+		// If it is a class prefix, check for deprecated class prefix also
+		if (InClassName.Len() > 12 && FCString::Strncmp(&(InClassName[1]), TEXT("DEPRECATED_"), 11) == 0)
+		{
+			PrefixSize = 12;
+		}
+		else
+		{
+			PrefixSize = 1;
+		}
+		break;
+	case TEXT('F'):
+	case TEXT('T'):
+		// Struct prefixes are also fine.
+		PrefixSize = 1;
+		break;
+	default:
+		PrefixSize = 0;
+		break;
+	}
+
+	return InClassName.RightChop(PrefixSize);
+}
+
+TArray<FAssetData> GetMatchingAssetsForSearchQuery(const FString& SearchValue)
+{
+	TArray<FSearchResult> OutItemsFound;
+	FStreamSearch StreamSearch(SearchValue);
+	while (!StreamSearch.IsComplete())
+	{
+		FFindInBlueprintSearchManager::Get().Tick(0.0);
+	}
+
+	StreamSearch.GetFilteredItems(OutItemsFound);
+
+	IAssetRegistry& AssetRegistry = FModuleManager::LoadModuleChecked<FAssetRegistryModule>(TEXT("AssetRegistry")).Get();
+
+	TArray<FAssetData> OutTargetAssets;
+	Algo::Transform(OutItemsFound, OutTargetAssets,
+		[&](const FSearchResult& Item)
+		{
+#if FILTER_ASSETS_BY_CLASS_PATH
+			return AssetRegistry.GetAssetByObjectPath(FSoftObjectPath(*Item->DisplayText.ToString()));
+#else
+			return AssetRegistry.GetAssetByObjectPath(*Item->DisplayText.ToString());
+#endif // FILTER_ASSETS_BY_CLASS_PATH
+		});
+
+	return OutTargetAssets;
+}
+
+TMap<FString, FAssetData> GetConfirmedAssets(
+	const FString& ClassName, const TArray<FAssetData>& InAssets)
+{
+	TMap<FString, FAssetData> OutResults;
+
+	AssetHelpers::ForEachAsset(InAssets,
+		[&](UBlueprintGeneratedClass* BlueprintClassName, const FAssetData AssetData)
+		{
+			auto It = Algo::FindByPredicate(BlueprintClassName->CalledFunctions,
+			[&](const UFunction* Fn)
+				{
+					return Fn->HasAnyFunctionFlags(EFunctionFlags::FUNC_Native)
+						&& Fn->GetOwnerClass()->GetName() != ClassName;
+				});
+
+			if (It != nullptr)
+			{
+				OutResults.Add(BlueprintClassName->GetName(), AssetData);
+			}
+		});
+
+	return OutResults;
+}
+using JsonWriter = TJsonWriter<TCHAR, TCondensedJsonPrintPolicy<TCHAR>>;
+
+static void SerializeBlueprintReference(
+	TSharedRef<JsonWriter>& Json, const FString& BlueprintClassName, const FAssetData& Asset)
+{
+	FString PackageFileName;
+	FString PackageFile;
+	FString PackageFilePath;
+	if (FPackageName::TryConvertLongPackageNameToFilename(Asset.GetPackage()->GetName(), PackageFileName) &&
+		FPackageName::FindPackageFileWithoutExtension(PackageFileName, PackageFile))
+	{
+		PackageFilePath = FPaths::ConvertRelativePathToFull(MoveTemp(PackageFile));
+	}
+	else
+	{
+		PackageFilePath = "[Invalid file path]";
```

Comment:
Maybe. I'm unsure because this is going to show up in the FAR window at some point and I think it does expects a file path and without it will just ignore the string altogether, so we can't really use this to show an error information.

I think a better path is to just emit an empty string and we use that on the VS side to display a better message if needed.